### PR TITLE
[coverage] More scapypipes tests

### PR DESCRIPTION
**Pull request by gpotter2**
This PR:
- adds more scapypipes tests
- fixes `_high_send` instead of `_send`
- tiny automaton speed-up & stabilize
- daemonize automaton threads to improve UTscapy closing

### CODE REVIEW

**Comment by gpotter2** [comment]
@p-l- This should be mergeable

**Comment by guedou** [comment]
@p-l- it looks good to me !

File: scapy/automaton.py [hunk 1]
```diff
@@ -92,15 +96,17 @@ def wait_return(self, callback):
...
+        _t.start()
         
     def call_release(self, arborted=False):
         """DEV: Must be call when the object becomes ready to read.
            Relesases the lock of _wait_non_ressources"""
         self.was_ended = arborted
         try:
             self.trigger.release()
-        except THREAD_EXCEPTION as e:
+        except (THREAD_EXCEPTION, AttributeError) as e:
```
Comment (p-l-) [inline scapy/automaton.py]: Please remove ` as e` as you don't need it.

File: scapy/automaton.py [hunk 2]
```diff
@@ -148,13 +150,20 @@ def __init__(self, inputs, remain):
     def process(self):
         """Entry point of SelectableSelector"""
         if WINDOWS:
+            select_inputs = []
             for i in self.inputs:
                 if not isinstance(i, SelectableObject):
-                    warning("Unknown ignored object type: " + type(i))
+                    warning("Unknown ignored object type: " + str(type(i)))
```
Comment (p-l-) [inline scapy/automaton.py]: Please use `warning("Unknown ignored object type: %s", type(i))`.

File: scapy/automaton.py [hunk 3]
```diff
@@ -148,13 +150,20 @@ def __init__(self, inputs, remain):
...
+                elif i.__selectable_force_select__:
+                    # Then use select.select
+                    select_inputs.append(i)
                 elif not self.remain and i.check_recv():
                     self.results.append(i)
                 else:
                     i.wait_return(self._exit_door)
+            if len(select_inputs) > 0:
```
Comment (p-l-) [inline scapy/automaton.py]: Please don't do that. `if select_inputs` is all you need.

File: scapy/automaton.py [hunk 4]
```diff
@@ -128,7 +130,7 @@ def _timeout_thread(self, remain):
             self._ended = True
             self._release_all()
 
-    def _exit_door(self,_input):
+    def _exit_door(self, _input):
```
Comment (p-l-) [inline scapy/automaton.py]: OK for this one, you're modifying the function, let's say you can do that ;-)

File: scapy/tools/UTscapy.py [hunk 5]
```diff
@@ -850,4 +850,4 @@ def main(argv):
     return glob_result
 
 if __name__ == "__main__":
-    exit(main(sys.argv[1:]))
+    sys.exit(main(sys.argv[1:]))
```
Comment (p-l-) [inline scapy/tools/UTscapy.py]: Why do you need that? (real question)

Comment (gpotter2) [reply scapy/tools/UTscapy.py]: Two reasons:
- while creating the PR, I noticed than when a thread would bug and never end, UTscapy would never close and stand by forever. Not safe for CI tests.
- quoting the doc:
https://docs.python.org/3/library/constants.html#exit
> The site module (which is imported automatically during startup, except if the -S command-line option is given) adds several constants to the built-in namespace. They are useful for the interactive interpreter shell and should not be used in programs.

**Comment by gpotter2** [comment]
Updated

**Pull request merged**